### PR TITLE
Fix bad build check for centipede

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

File: infra/base-images/base-runner/test_all.py [hunk 1]
```diff
@@ -220,6 +218,12 @@ def test_all(out, allowed_broken_targets_percentage):
     print('ERROR: No fuzz targets found.')
     return False
 
+  if centipede_needs_auxiliaries():
```
Comment (jonathanmetzman) [inline infra/base-images/base-runner/test_all.py:221]: I don't really like this check. Why do we have it at all @Alan32Liu

**Comment by jonathanmetzman** [comment]
Related: https://github.com/google/oss-fuzz/issues/9299

**Comment by jonathanmetzman** [comment]
/gcbrun trial_build.py all --sanitizer none address --fuzzing-engine centipede

**Comment by jonathanmetzman** [comment]
TBR

File: infra/base-images/base-runner/test_all.py (hunk 1 above)
Comment (Alan32Liu) [reply infra/base-images/base-runner/test_all.py:221]: When I wrote this code, `Centipede` requires an extra non-sanitized binary if fuzzing sanitizer binaries.
That non-sanitized binary is the auxiliary here.

We can simplify this design later as `Centipede` can work without that non-sanitized auxiliary binary, IIUC.
But I presume that is not urgent?

Comment (jonathanmetzman) [reply infra/base-images/base-runner/test_all.py:221]: Still not sure if it's needed but if it matches the behavior of production i presume it's fine?

Comment (Alan32Liu) [reply infra/base-images/base-runner/test_all.py:221]: I think it's better to keep it for now, and then remove it when we want to adapt to `Centipede`'s new feature that does not want to use the unsanitized binary.

**Pull request merged**